agenta-backend/agenta_backend/services/app_manager.py

```diff
@@ -558,4 +610,110 @@ async def add_variant_based_on_image(
         base_name=base_name,
     )
     logger.debug("End: Successfully created db_app_variant: %s", db_app_variant)
+
+    return db_app_variant
+
+
+async def add_variant_based_on_url(
+    app: AppDB,
+    project_id: str,
+    variant_name: str,
+    url: str,
+    user_uid: str,
+    base_name: Optional[str] = None,
+    config_name: str = "default",
+) -> AppVariantDB:
+    """
+    Adds a new variant to the app based on the specified URL.
+
+    Args:
+        app (AppDB): The app to add the variant to.
+        project_id (str): The ID of the project.
+        variant_name (str): The name of the new variant.
+        url (str): The URL to use for the new variant.
+        base_name (str, optional): The name of the base to use for the new variant. Defaults to None.
+        config_name (str, optional): The name of the configuration to use for the new variant. Defaults to "default".
+        user_uid (str): The UID of the user.
+
+    Returns:
+        AppVariantDB: The newly created app variant.
+
+    Raises:
+        ValueError: If the app variant or URL is None, or if an app variant with the same name already exists.
+        HTTPException: If an error occurs while creating the app variant.
+    """
+
+    logger.debug("Start: Creating app variant based on url")
+
+    logger.debug("Validating input parameters")
+    if app in [None, ""] or variant_name in [None, ""] or url in [None, ""]:
+        raise ValueError("App variant, variant name, or URL is None")
+
+    logger.debug("Parsing URL")
+    parsed_url = urlparse(url).geturl()
+
+    logger.debug("Checking if app variant already exists")
+    variants = await db_manager.list_app_variants_for_app_id(
+        app_id=str(app.id),
+        project_id=project_id,
+    )
+
+    already_exists = any(av for av in variants if av.variant_name == variant_name)  # type: ignore
+    if already_exists:
+        logger.error("App variant with the same name already exists")
+        raise ValueError("App variant with the same name already exists")
+
+    logger.debug("Retrieving user and image objects")
+    user_instance = await db_manager.get_user(user_uid)
+
+    # Create config
+    logger.debug("Creating config")
+    config_db = await db_manager.create_new_config(
+        config_name=config_name, parameters={}
+    )
+
+    # Create base
+    logger.debug("Creating base")
+    if not base_name:
+        base_name = variant_name.split(".")[
+            0
+        ]  # TODO: Change this in SDK2 to directly use base_name
+    db_base = await db_manager.create_new_variant_base(
+        app=app,
+        project_id=project_id,
+        base_name=base_name,  # the first variant always has default base
```

Comment:
```suggestion
        base_name=base_name, 
```

Comment not true, no? The first variant has always the `default` config not base